recurly.d.ts

```diff
@@ -0,0 +1,826 @@
+// Type definitions for recurly
+// Project: recurly
+// Definitions by: Benjamin Eckel <dx@recurly.com>
+
+export as namespace recurly;
+
+export interface Site {
+  id: string | null;
```

Comment:
@wcauchois What do you think about these changes? Changed from optional to explicit `type | null` seems to be working on my end.

Reply:
looks good!